components/datetime/src/neo.rs

```diff
@@ -563,7 +563,7 @@ where
         datetime: &I,
     ) -> Result<FormattedDateTime, crate::MismatchedCalendarError>
     where
-        I: ?Sized + IsInCalendar + AllInputMarkers<FSet>,
+        I: ?Sized + InCalendar + AllInputMarkers<FSet>,
```

Comment:
optional: how about InSameCalendar to (a) make it more meaningful and (b) make it more parallel with InFixedCalendar?

Reply:
Done

Justification: the fully qualified names have good parallelism

- `DateTimeFormatter::format_same_calendar` => `InSameCalendar`
- `FixedCalendarDateTimeFormatter::format` => `InFixedCalendar`

Also, "InCalendar" is too short and could mean something else in the future.